pkg/storage/db.go

```diff
@@ -199,6 +208,79 @@ func (db *DB) ping() error {
 	return nil
 }
 
+// GeneratePartition finds the latest partition, and generate the next partition based on the previous partions time range
+func (db *DB) GeneratePartition(ctx context.Context) error {
+	stmt := "SELECT partition_end FROM partitions ORDER BY partition_begin DESC LIMIT 1"
```

Comment:
Suggestion: I'm reading this as `Give me the partition_end for the latest partition_create`. Granted, if we're preserving consistency of the data then we won't have a problem but it's more technically correct to ask `Give me the latest partition_end` if we are guaranteeing a fast-forward only model.